packages/mobile/src/identity/matchmaking.ts

```diff
@@ -12,12 +12,16 @@ import { NumberToRecipient } from 'src/recipients/recipient'
 import Logger from 'src/utils/Logger'
 import { getAuthSignerForAccount } from 'src/web3/dataEncryptionKey'
 import { getConnectedUnlockedAccount } from 'src/web3/saga'
+import { mtwAddressSelector } from 'src/web3/selectors'
 
 const TAG = 'identity/matchmaking'
 
 // Uses the phone number privacy service to find mutual matches between Celo users
 export function* fetchContactMatches(e164NumberToRecipients: NumberToRecipient) {
-  const account: string = yield call(getConnectedUnlockedAccount)
+  const walletAddress: string = yield call(getConnectedUnlockedAccount)
+  const mtwAddress: string | null = yield select(mtwAddressSelector)
+  const accountAddress: string = mtwAddress || walletAddress
```

Comment:
Good point about isolating the disambiguation. Probably too soon to integrate it into existing `getConnected*` functions because we don't want to do this everywhere just yet. I repeat this logic at least a few times (and will probably add more in subsequent PRs) so a new getter function makes a lot of sense.

As for terminology, what I was going for was consistency with how these are referred to on the contracts side where a user can have both an `accountAddress` and `walletAddress`. This is generally supposed to be a parent (account) child (wallet) relationship where the two addresses can be the same but don't have to be. Am I misunderstanding the existing terminology?

Reply:
Yeah +1 for account and wallet address